File: backend/gb/src/main/java/com/gamebox/gb/domain/dtos/profile/ProfileDashboardResponse.java
package com.gamebox.gb.domain.dtos.profile;

import com.gamebox.gb.domain.dtos.favorite.FavoriteResponse;
import com.gamebox.gb.domain.dtos.played.PlayedResponse;
import com.gamebox.gb.domain.dtos.review.ReviewSearchResponse;
import com.gamebox.gb.domain.dtos.wishlist.WishlistSearchResponse;

import java.util.List;

public record ProfileDashboardResponse(
        ProfileResponse profile,
        List<PlayedResponse> playedGames,
        List<FavoriteResponse> favorites,
        List<WishlistSearchResponse> wishlistGames,
        List<ReviewSearchResponse> reviews
) {}

File: frontend/src/models/profile/ProfileDashboardResponse.ts
import { ProfileResponse } from "./ProfileResponse";
import { PlayedResponse } from "@/models/played/PlayedResponse";
import { FavoriteResponse } from "@/models/favorite/FavoriteResponse";
import { WishlistSearchResponse } from "@/models/wishlist/WishlistSearchResponse";
import { ReviewSearchResponse } from "@/models/review/ReviewSearchResponse";

export type ProfileDashboardResponse = {
  profile: ProfileResponse;
  playedGames: PlayedResponse[];
  favorites: FavoriteResponse[];
  wishlistGames: WishlistSearchResponse[];
  reviews: ReviewSearchResponse[];
};